app/src/components/Market/OrderCard/components/Swap/Swap.tsx

```diff
@@ -96,6 +96,15 @@ const Swap: React.FC = () => {
       tokenAddress = underlyingToken.address
       balance = underlyingToken
       break
+    case Operation.WRITE:
+      title = {
+        text: 'Write Options',
+        tip:
+          'Underwrite long option tokens with an underlying token deposit, and sell them for premiums denominated in underlying tokens.',
+      }
+      tokenAddress = item.entity.address //underlyingToken.address FIX: double approval
```

Comment:
Just to confirm, we need double approvals in Swap?  Should be fairly simple extension of my solution for the withdraw double approval.

Reply:
Yessir